Wikipedia/Code/SectionEditorViewController.swift

```diff
@@ -95,19 +97,30 @@ class SectionEditorViewController: ViewController {
         navigationItemController.delegate = self
         
         loadEditNotices()
-        loadWikitext { [weak self] blockedError in
+        loadWikitext { [weak self] resultError in
             
             guard let self else {
                 return
             }
             
-            if let blockedError {
-                self.presentErrorMessage(blockedError: blockedError)
-                
-                DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + 0.2) { // helps prevent flash as wikitext is loaded
-                    self.configureWebView(readOnly: true)
+            if let error = resultError {
+                if error.code.contains("protectedpage") {
+                    self.presentProtectedPageWarning(error: error)
+                    DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + 0.2) { // helps prevent flash as wikitext is loaded
+
+                        self.configureWebView(readOnly: true)
+                    }
+                } else {
+                    self.presentErrorMessage(blockedError: error)
```

Comment:
I haven't tested, but you might need to move this check to before checking protection levels - it looks like from the task description they want blocked messages to take precedence.